### PR TITLE
examples/{tc,xdp}*: Use network-types crate instead of aya-tool bindings

### PR DESCRIPTION
**Pull request by vadorovsky**
Network-related structures don't change across kernel versions, so using BTF to generate them is an overkill. Instead of using aya-tool, we decided to publish a crate called network-types, which provides definitions of network headers (L2, L3 and L4) which are suitable to use as compatible with ones provided by the kernel.

Signed-off-by: Michal Rostecki <vadorovsky@gmail.com>

### CODE REVIEW

File: docs/book/start/hello-xdp.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 !!! example "Source Code"
 
-    Full code for the example in this chapter is availble [here](https://github.com/aya-rs/book/tree/main/examples/myapp-01)
+    Full code for the example in this chapter is availble [here](https://github.com/aya-rs/book/tree/main/examples/xdp-hello)
```
Comment (shaneutt) [inline docs/book/start/hello-xdp.md:5]:
```suggestion
    Full code for the example in this chapter is available [here](https://github.com/aya-rs/book/tree/main/examples/xdp-hello)
```

File: docs/book/start/logging-packets.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ Let's expand this program to log the traffic that is being permitted in the user
 
 !!! example "Source Code"
 
-    Full code for the example in this chapter is availble [here](https://github.com/aya-rs/book/tree/main/examples/myapp-02)
+    Full code for the example in this chapter is availble [here](https://github.com/aya-rs/book/tree/main/examples/xdp-log)
```
Comment (shaneutt) [inline docs/book/start/logging-packets.md:9]:
```suggestion
    Full code for the example in this chapter is available [here](https://github.com/aya-rs/book/tree/main/examples/xdp-log)
```

File: docs/book/start/logging-packets.md [hunk 3]
```diff
@@ -87,73 +87,50 @@ The data structure that we'll need to send information to user-space will need t
...
-Lets use `xtask` to automate this so we can easily reproduce this file in future.
-
-We'll add the following code
-
-=== "xtask/src/codegen.rs"
-
-    ```rust linenums="1"
-    --8<-- "examples/myapp-02/xtask/src/codegen.rs"
-    ```
+Let's add it to our eBPF crate by adding a dependency on `network-types = "0.0.2"`
```
Comment (shaneutt) [inline docs/book/start/logging-packets.md:103]: These kinds of versions can get lost in time and cause issues for readers. Has there been any previous thoughts on doing something like templating from the `Cargo.toml` data in the relevant examples? Not at all something to hold up this PR for of course, but if there's any interest maybe I can throw together an issue for it.

**Comment by alessandrod** [comment]
I think this is good, but the network-types crate needs a bit more work before we can merge

File: examples/tc-egress/tc-egress-ebpf/src/main.rs [hunk 4]
```diff
@@ -1,31 +1,20 @@
...
-use bindings::{ethhdr, iphdr};
-
-#[map(name = "EVENTS")]
-static mut EVENTS: PerfEventArray<PacketLog> =
-    PerfEventArray::with_max_entries(1024, 0);
+use aya_log_ebpf::info;
+use network_types::{
+    l2::ethernet::{EthHdr, ETH_HDR_LEN},
+    l3::{ip::Ipv4Hdr, L3Protocol},
+};
```
Comment (shaneutt) [inline examples/tc-egress/tc-egress-ebpf/src/main.rs:14]: :+1:

**Review by shaneutt — COMMENTED** [review]
Looks good, I'm really glad to see this crate available thanks for the effort :vulcan_salute:

File: docs/book/start/logging-packets.md (hunk 3 above)
Comment (vadorovsky) [reply docs/book/start/logging-packets.md]: Good point. I removed the version from the sentence, we show the whole `Cargo.toml` content below anyway, so I think that should suffice.

File: docs/book/start/hello-xdp.md (hunk 1 above)
Comment (vadorovsky) [reply docs/book/start/hello-xdp.md]: done

File: docs/book/start/logging-packets.md (hunk 3 above)
Comment (vadorovsky) [reply docs/book/start/logging-packets.md]: done

File: docs/book/start/logging-packets.md (hunk 2 above)
Comment (vadorovsky) [reply docs/book/start/logging-packets.md]: done

**Comment by vadorovsky** [comment]
Let's wait for https://github.com/vadorovsky/network-types/pull/3 to get merged before this one

**Comment by vadorovsky** [comment]
I updated the PR to use the newest network-types.

**Review by dave-tucker — APPROVED** [review]
LGTM

**Review by shaneutt — APPROVED** [review]

**Pull request merged**